components/suggestion/switch_channel_provider.jsx

```diff
@@ -229,11 +229,11 @@ export default class SwitchChannelProvider extends Provider {
         }
 
         const users = Object.assign([], searchProfiles(state, channelPrefix, false)).concat(usersFromServer.users);
-        const channels = getChannelsInCurrentTeam(state).concat(getGroupChannels(state)).concat(channelsFromServer);
+        const channels = getChannelsInCurrentTeam(state).concat(getDirectChannels(state)).concat(channelsFromServer);
```

Comment:
Won't this only get direct channels that already exist? One of the features of the channel switcher is letting you open a DM with a user you've never DMed before

Reply:
I can test that but the reality is that before this change it was using the selector for only getting group messages and I changed it to use the selector that returns direct messages and group messages, so it should not have affected the ability to open DM's with those who you didn't had a DM before

Reply:
What can be an issue is that I removed the loop through the users and maybe that can prevent the ability to open DM's with others, I'll be checking that